File: frontend/src/services/orderService.js
import api from './api';

export const checkout = async (payload) => {
  const response = await api.post('/checkout', payload);
  return response.data.data;
};

export const getOrders = async () => {
  const response = await api.get('/orders');
  return response.data.data;
};

export const getOrderById = async (orderId) => {
  const response = await api.get(`/orders/${orderId}`);
  return response.data.data;
};

File: backend/src/main/java/com/shoppingcart/service/OrderService.java
package com.shoppingcart.service;

import com.shoppingcart.dto.OrderDtos;

import java.util.List;

public interface OrderService {

    OrderDtos.OrderResponse checkout(OrderDtos.CheckoutRequest request);

    List<OrderDtos.OrderResponse> getOrders();

    OrderDtos.OrderResponse getOrderById(String orderId);
}
